met/src/basic/vx_util/RectangularTemplate.cc

```diff
@@ -52,82 +29,63 @@
 #include <RectangularTemplate.h>
 #include <GridOffset.h>
 #include <GridTemplate.h>
+
 using namespace std;
 
-/**********************************************************************
- * Constructor
- */
-
-RectangularTemplate::RectangularTemplate(int height, int width) :
-  GridTemplate(),
-  _height(height),
-  _width(width)
-{
-
-  /*
-    CONVENTION:
-    If width is even, we push the center to the lower left corner
-    and have the "extra" in the right & above.
-  */
-
-  bool evenWidth = ((width % 2) == 0);
-  bool evenHeight = ((height % 2) == 0);
-
-  // equivalent of w/2 for even & (w-1)/2 for odd
-  int halfwidth = width / 2;
-  int halfheight = height / 2;
-
-  int xmin = halfwidth;
-  int ymin = halfheight;
-
-  if (evenWidth){
-    xmin--;
-  }
-  if (evenHeight){
-    ymin--;
-  }
-
-  // Create the offsets list
-  for (int y = -ymin; y <= halfheight; y++)
-  {
-    for (int x = -xmin; x <= halfheight; x++)
-    {
-      _addOffset(x, y);
-    } /* endfor x = 0 */
-  } /* endfor y = 0*/
-
-  _setEdgeOffsets();
-}
+///////////////////////////////////////////////////////////////////////////////
 
+RectangularTemplate::RectangularTemplate(int height, int width, bool wrap_lon) :
+   GridTemplate(), _height(height), _width(width) {
 
-/**********************************************************************
- * Destructor
- */
+   _wrapLon = wrap_lon;
 
-RectangularTemplate::~RectangularTemplate(void)
-{
-  // Do nothing
-}
+   //
+   // CONVENTION:
+   // If width is even, we push the center to the lower left corner
+   // and have the "extra" in the right & above.
+   //
 
+   bool evenWidth = ((width % 2) == 0);
+   bool evenHeight = ((height % 2) == 0);
 
-/**********************************************************************
- * printOffsetList() - Print the offset list to the given stream.  This
- *                     is used for debugging.
- */
+   // equivalent of w/2 for even & (w-1)/2 for odd
+   int halfwidth = width / 2;
+   int halfheight = height / 2;
 
-void RectangularTemplate::printOffsetList(FILE *stream)
-{
-  fprintf(stream, "\n\n");
-  fprintf(stream, "Rectangular template:");
-  fprintf(stream, "    height = %d\n", _height);
-  fprintf(stream, "    width = %d\n", _width);
+   int xmin = halfwidth;
+   int ymin = halfheight;
 
-  fprintf(stream, " grid points:\n");
+   if(evenWidth)  xmin--;
+   if(evenHeight) ymin--;
+
+   // Create the offsets list
+   for(int y = -ymin; y <= halfheight; y++) {
+      for(int x = -xmin; x <= halfheight; x++) {
```

Comment:
halfheight --> halfwidth

Reply:
Thanks Howard. You're right that should be halfwidth instead, and I just made that [change](https://github.com/dtcenter/MET/commit/10d726f2d202d5aeed28deaea22eb09f07c4398f). This change has no impact on results since MET only uses the rectangular template to create squares where height = width. But its still good to correct.